src/lib/badge/badge.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, Input, Renderer2, ElementRef} from '@angular/core';
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+import {ThemePalette} from '@angular/material/core';
+
+let nextId = 0;
+
+/** Directive to display a text badge. */
+@Directive({
```

Comment:
Description should talk a bit about how this is typically used.

Reply:
None of the other components have this, can you provide an example of what'd you like to see?